src/wled2.ts

```diff
@@ -134,6 +136,29 @@ export = (RED: Red): void => {
     this.send(msg);
   }
 
+  async function getEffectFromString(address: string, payloadEffect: string | number | null, configEffect: string | number | null) {
+    if (typeof payloadEffect === 'number') return payloadEffect;
+    try {
+      const result = await fetch(`http://${address}/json/effects`);
+      if (!result.ok) {
+        return -1;

```

Comment:
Instead of returning -1 on error it should throw an exception or return null. The rest of the code throws exceptions on errors so that's probably the right way to do it here as well.

The advantage to returning null would be the ability to use ?? and ?. operators elsewhere in code when dealing with the returned value from this fuction.